### PR TITLE
Do NOT clone input vars if op NOT in pserver_program

### LINKED ISSUE #11259: fluid_benchmark 运行加入----use_reader_op参数，启动psserver失败
测试pr #11121 启动多机训练，启动psserver脚本如下，出错：
export FLAGS_fraction_of_gpu_memory_to_use=0.0 
export PADDLE_TRAINING_ROLE=PSERVER 
export PADDLE_PSERVER_PORT=5002 
export PADDLE_PSERVER_IPS=10.255.123.15,10.255.123.16 
export PADDLE_TRAINERS=2 
export PADDLE_CURRENT_IP=10.255.123.15 
export PADDLE_TRAINER_ID=1
python fluid_benchmark.py --model resnet  \
--data_set flowers --use_reader_op --memory_optimize \
--iterations 20 \
--device GPU --gpus 8 --batch_size 64 \
--pass_num 1 --data_path recordio --no_test \
--update_method pserver
报错如下：
image

去掉--use_reader_op参数后可以正常启动

### PR DESCRIPTION
**Pull request by velconia**
Only the inputs vars of ops in Pserver program will be cloned, other input vars will NOT take into account.

this close #11259

### CODE REVIEW

File: paddle/fluid/string/printf.h [hunk 1]
```diff
@@ -84,7 +84,7 @@ void Fprintf(std::ostream& out, const char* fmt, const Args&... args) {
 }
 
 template <typename... Args>
-std::string Sprintf(const char* fmt = "", const Args&... args) {
```
Comment (seiriosPlus) [inline paddle/fluid/string/printf.h]: 把 Sprintf 这个单独提个PR？ 跟修复的问题无关

Comment (velconia) [reply paddle/fluid/string/printf.h]: 好的

Comment (typhoonzero) [reply paddle/fluid/string/printf.h]: https://github.com/PaddlePaddle/Paddle/pull/11873 包含了修复

Comment (velconia) [reply paddle/fluid/string/printf.h]: Done, 我把修复移到另一个PR并且merge了

**Review by typhoonzero — APPROVED** [review]
LGTM++

**Pull request merged**